### PR TITLE
doc: Added Build Infra Doc

### PR DESCRIPTION
**Pull request by ashokdevtron**
- Build Infra Doc added with snapshots
- Covered single and bulk actions
- Added link to GitHub issue for more infra config support
- CPU and Memory unit ref included in Extras

### CODE REVIEW

File: docs/user-guide/global-configurations/build-infra.md [hunk 1]
```diff
@@ -0,0 +1,157 @@
+# Build Infra
+
+## Introduction
+
+The [CI process](../creating-application/workflow/ci-pipeline.md) involves activities that require infra resources such as CPU, memory (RAM), and many more. The amount of resources depends on the complexity of the application. In other words, large applications require more resources compared to small applications.
```
Comment (uxarya-d) [inline docs/user-guide/global-configurations/build-infra.md:5]: The amount of resources required depends…

File: docs/user-guide/global-configurations/build-infra.md [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+![Figure 15: Edit and Delete Icons](https://devtron-public-asset.s3.us-east-2.amazonaws.com/images/global-configurations/build-infra/edit-delete-icons-v2.jpg)
+
+If you delete a profile attached to one or more applications, the [default profile](#default-profile) will apply from the next build.
+
+![Figure 16: Confirm Profile Deletion](https://devtron-public-asset.s3.us-east-2.amazonaws.com/images/global-configurations/build-infra/delete-dialog.jpg)
+
+
+### Need More Options?
+
+If you need extra control on the build infra configuration apart from CPU and memory, feel free to open a [GitHub issue](https://github.com/devtron-labs/devtron/issues) for us to help you.
```
Comment (uxarya-d) [inline docs/user-guide/global-configurations/build-infra.md:118]: … from CPU, memory and build timeout, feel…

File: docs/user-guide/global-configurations/build-infra.md [hunk 3]
```diff
@@ -0,0 +1,157 @@
+# Build Infra
+
+## Introduction
+
+The [CI process](../creating-application/workflow/ci-pipeline.md) involves activities that require infra resources such as CPU, memory (RAM), and many more. The amount of resources depends on the complexity of the application. In other words, large applications require more resources compared to small applications.
+
+Therefore, applying a common infra configuration to all applications is not optimal. Since resources incur heavy costs, it's wise to efficiently allocate resources (*not more, not less*).
```
Comment (uxarya-d) [inline docs/user-guide/global-configurations/build-infra.md:7]: Please include that the build (ci-runner) pod will be scheduled on an available node (considering applied taints and tolerations) in the cluster on which 'Devtron' is installed.

**Review by uxarya-d — CHANGES_REQUESTED** [review]
Added comments

File: docs/user-guide/global-configurations/build-infra.md (hunk 3 above)
Comment (ashokdevtron) [reply docs/user-guide/global-configurations/build-infra.md]: I will include this line as it is for now...😅
I wish to know more about this build pod technicality, please let's discuss this in our next conversation.

**Review by uxarya-d — APPROVED** [review]

**Pull request merged**